signer/src/utxo.rs

```diff
@@ -461,28 +478,79 @@ impl<'a> UnsignedTransaction<'a> {
             })
             .collect::<Result<_, _>>()?;
 
-        // Combine the them all together to get an ordered list of taproot
+        // Combine them all together to get an ordered list of taproot
         // signature hashes.
         Ok(SignatureHashes {
             signers: signer_sighash,
             deposits: deposit_sighashes,
         })
     }
 
+    /// Compute the sum of the input amounts of the transaction
+    pub fn input_amounts(&self) -> u64 {
+        self.requests
+            .iter()
+            .filter_map(Request::as_deposit)
+            .map(|dep| dep.amount)
+            .chain([self.signer_utxo.utxo.amount])
+            .sum()
+    }
+
+    /// Compute the sum of the output amounts of the transaction.
+    pub fn output_amounts(&self) -> u64 {
+        self.tx.output.iter().map(|out| out.value.to_sat()).sum()
+    }
+
     /// Compute the fee that each deposit and withdrawal request must pay
-    /// for the transaction given the fee rate
+    /// for the transaction given the fee rate and the fees paid last time
+    /// if this is a replace-by-fee (RBF) transaction.
     ///
     /// If each deposit and withdrawal associated with this transaction
     /// paid the fees returned by this function then the fee rate for the
-    /// entire transaction will be at least as much as the fee rate.
+    /// entire transaction will be at least as much as the target fee rate.
+    /// Moreover, both the transaction's total fee and fee rate will be
+    /// greater than the `last_fee` if present.
     ///
-    /// Note that each deposit and withdrawal pays an equal amount for the
-    /// transaction. To compute this amount we divide the total fee by the
-    /// number of requests in the transaction.
-    fn compute_request_fee(tx: &Transaction, fee_rate: u64) -> u64 {
-        let tx_fee = tx.vsize() as u64 * fee_rate;
-        let num_requests = (tx.input.len() + tx.output.len()).saturating_sub(2) as u64;
-        tx_fee.div_ceil(num_requests)
+    /// ## Notes
+    ///
+    /// Each deposit and withdrawal pays an equal amount for the transaction.
+    /// To compute this amount we divide the total fee by the number of
+    /// requests in the transaction.
+    ///
+    /// Here are the fee related requirements for a replace-by-fee as
+    /// described in BIP-125:
+    ///
+    /// 3. The replacement transaction pays an absolute fee of at least the
+    ///    sum paid by the original transactions.
+    /// 4. The replacement transaction must also pay for its own bandwidth
+    ///    at or above the rate set by the node's minimum relay fee setting.
+    ///    For example, if the minimum relay fee is 1 satoshi/byte and the
+    ///    replacement transaction is 500 bytes total, then the replacement
+    ///    must pay a fee at least 500 satoshis higher than the sum of the
+    ///    originals.
```

Comment:
I know this comes directly from the BIP, but it is very unclear to me what "sum of the originals" refer to here. This sounds like it should just be the fee of the replaced transaction, but then I don't know what the sum is over. Do you know what it refers to?

Reply:
It's not clear to me either but I suspect it has to do with transactions that depend on the transaction that we are replacing. This is something I plan on finding out when I test replacing a transaction package.